core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -102,37 +106,46 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
       deleteFiles(io, manifestsToDelete);
     }
 
-    Tasks.foreach(Iterables.transform(manifestsToDelete, ManifestFile::path))
-        .executeWith(ThreadPools.getWorkerPool())
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure((manifest, exc) -> LOG.warn("Delete failed for manifest: {}", manifest, exc))
-        .run(io::deleteFile);
-
-    Tasks.foreach(manifestListsToDelete)
-        .executeWith(ThreadPools.getWorkerPool())
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure((list, exc) -> LOG.warn("Delete failed for manifest list: {}", list, exc))
-        .run(io::deleteFile);
-
-    Tasks.foreach(
-            Iterables.transform(metadata.previousFiles(), TableMetadata.MetadataLogEntry::file))
-        .executeWith(ThreadPools.getWorkerPool())
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure(
-            (metadataFile, exc) ->
-                LOG.warn("Delete failed for previous metadata file: {}", metadataFile, exc))
-        .run(io::deleteFile);
-
-    Tasks.foreach(metadata.metadataFileLocation())
-        .noRetry()
-        .suppressFailureWhenFinished()
-        .onFailure(
-            (metadataFile, exc) ->
-                LOG.warn("Delete failed for metadata file: {}", metadataFile, exc))
-        .run(io::deleteFile);
+    if (io instanceof SupportsBulkOperations) {
+      SupportsBulkOperations bulkFileIO = (SupportsBulkOperations) io;
+      bulkFileIO.deleteFiles(Iterables.transform(manifestsToDelete, ManifestFile::path));
+      bulkFileIO.deleteFiles(manifestListsToDelete);
+      bulkFileIO.deleteFiles(
+          Iterables.transform(metadata.previousFiles(), TableMetadata.MetadataLogEntry::file));
+      bulkFileIO.deleteFiles(ImmutableList.of(metadata.metadataFileLocation()));
+    } else {
```

Comment:
to minimize git line changes, can we do

```
if (io instanceof SupportsBulkOperations) {
  ...
  return
}

...
```

Reply:
Yeah this way it is also easier to review, done.